packages/react-instantsearch-core/src/lib/useInstantSearchApi.tsx

```diff
@@ -50,15 +55,31 @@ export type InternalInstantSearch<
   _preventWidgetCleanup?: boolean;
 };
 
+const InstantSearchInitialResults = Symbol.for('InstantSearchInitialResults');
+declare global {
+  interface Window {
+    [InstantSearchInitialResults]?: InitialResults;
+  }
+}
+
 export function useInstantSearchApi<TUiState extends UiState, TRouteState>(
   props: UseInstantSearchApiProps<TUiState, TRouteState>
 ) {
   const forceUpdate = useForceUpdate();
   const serverContext = useInstantSearchServerContext<TUiState, TRouteState>();
   const serverState = useInstantSearchSSRContext<TUiState, TRouteState>();
-  const initialResults = serverState?.initialResults;
+  const { promiseRef, insertHTML } = useRSCContext();
+  let initialResults =
+    serverState?.initialResults ||
+    (typeof window !== 'undefined'
+      ? window[InstantSearchInitialResults]
+      : undefined);
   const prevPropsRef = useRef(props);
 
+  if (Array.isArray(initialResults)) {
```

Comment:
Not sure why their `insertHTML` function inserts it twice, one in `<head>` and one in `<body>`.
Also from the docs they say it should be in `layout.js` but seems like it works in a `page.js` file ?